system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp

```diff
@@ -1081,6 +1081,151 @@ bool AzureLogAnalyticsCurlClient::processSearchJsonResp(LogQueryResultDetails &
+            report.appendf(", \"SpanID\": { \"ColName\": \"%s\", \"Source\": \"%s\"}", m_spanSearchColName.str(), m_spanIndexSearchPattern.str());
+            report.appendf(", \"Host\": { \"ColName\": \"%s\", \"Source\": \"%s\"}", m_hostSearchColName.str(), m_hostIndexSearchPattern.str());
+            report.append(" }"); //close logmaps
+            report.append(" }"); //close plugin
+        }
+
+        if (options.IncludeServerInternals)
+        {
+            report.append(", \"Server\": { }");
+        }
+
+        report.append(" }"); //close internals
+        if (options.IncludeSampleQuery)
+        {
+            report.append(", \"SampleTokenRequest\": { ");
+            try
+            {
+                
+                StringBuffer token;
+                requestLogAnalyticsAccessToken(token, m_aadClientID, m_aadClientSecret, m_aadTenantID); //throws if issues encountered
+
+                if (token.isEmpty())
+                    report.append("\"Error\": \"Empty token received\"");
+
+            }
+            catch(IException * e)
+            {
+                StringBuffer description;
+                e->errorMessage(description);
+                report.appendf("\"Error\": \"Exception while requesting token (%d) - %s\"", e->errorCode(), description.str());
+                e->Release();
+            }
+            catch(...)
+            {
+                report.append("\"Error\": \"Unknown exception while requesting token\"");
+            }
+            report.append(" }"); //close sample token request
+
+            report.append(", \"SampleQuery\": { ");
+            try
+            {
+                report.appendf("\"Query\": { \"LogFormat\": \"JSON\",");
+                LogAccessLogFormat outputFormat = LOGACCESS_LOGFORMAT_json;
+                LogAccessConditions queryOptions;
+
+                report.appendf("\"Filter\": {\"type\": \"byWildcard\", \"value\": \"*\" },");
+                queryOptions.setFilter(getWildCardLogAccessFilter("*"));
+
+                struct LogAccessTimeRange range;
+                CDateTime endtt;
+                endtt.setNow();
+                range.setEnd(endtt);
+                StringBuffer endstr;
+                endtt.getString(endstr);
+
+                CDateTime startt;
+                startt.setNow();
+                startt.adjustTimeSecs(-60); //an hour ago
+                range.setStart(startt);
+
+                StringBuffer startstr;
+                startt.getString(startstr);
+                report.appendf("\"TimeRange\": {\"Start\": \"%s\", \"End\": \"%s\" },", startstr.str(), endstr.str());
+
+                queryOptions.setTimeRange(range);
+                queryOptions.setLimit(5);
+                report.appendf("\"Limit\": \"5\" }, ");
+
+                StringBuffer queryString, queryIndex;
+                populateKQLQueryString(queryString, queryIndex, queryOptions);
+                
+                StringBuffer encodedValue;
```

Comment:
Is this the _only_ string that needs to be encoded, or are there other cases where user input could be part of any value returned in the report? Would it be a good idea to encode all strings returned?